zerver/actions/user_groups.py

```diff
@@ -245,7 +244,7 @@ def do_send_delete_user_group_event(realm: Realm, user_group_id: int, realm_id:
     send_event(realm, event, active_user_ids(realm_id))
 
 
-def check_delete_user_group(user_group_id: int, *, acting_user: UserProfile) -> None:
-    user_group = access_user_group_by_id(user_group_id, acting_user, for_read=False)
+def check_delete_user_group(user_group: UserGroup, *, acting_user: UserProfile) -> None:
+    user_group_id = user_group.id
```

Comment:
We make the caller access the user group prior to deletion so that all the locking and access checks are done in the view functions.

Reply:
Yeah I think we need to do it here, since if we add a separate type of code (say, a management command syncing groups from LDAP), that will need to follow the locking protocol.